skycoin/skywire
# Fix/shutdown data races

Did you run `make format && make check`?

Fixes #774

 Changes:	
- Fix data races when manually shutting down visor

How to test this PR:
Run visor, wait until initialization finishes and then hit Ctrl+C

Note: I removed synchronization in router shutdown: [commit](https://github.com/skycoin/skywire/commit/15e0ebca5c4405f3996c7d5194af69a45966ffbb). There was a waitgroup that waits until router `serveSetup` method exists completely  that was causing data race. Apparently, you cannot use `Add` and `Wait` wg methods in different goroutines. Not sure if we need this synchronization there, but should be possible to achieve similar result with a channel. 

Another issue is that visor exits with status 1, i.e. error, so the shutdown process is not considered successful. This is the observable behavior on develop and wasn't introduced by this PR.
